index.html

```diff
@@ -2449,6 +2695,254 @@ <h3 id=new-session><dfn data-lt="new sessions|creating a new session">New Sessio
+     with a <a>session creation job id</a> matching <var>session creation job id</var>.
+
+    <li><p>If <var>session creation job</var> is <a><code>null</code></a>,
+     return <a>error</a> with <a>error code</a> <a>invalid session creation job id</a>.
+
+    <li><p>Let <var>body</var> be a <a>JSON clone</a> of <var>session creation job</var>.
+
+    <li><p>If <var>event</var> is defined:
+
+     <ol>
+      <li><p>Let <var>filtered events</var> be an empty JSON <a>List</a>.
+
+      <li><p>Let <var>found</var> be false.
+
+      <li><p>Let <var>events</var> be the value of <a>getting a property</a> named <var>events</var> from <var>body</var>.
+
+      <li><p>For each <var>event object</var> in <var>events</var> in reverse order:
+
+      <ol>
+       <li><p>If <var>found</var> is true, then break.
+
+       <li><p>Let <var>event id</var> be the value of <a>getting a property</a> named <var>eventId</var> from <var>event object</var>.
+
+       <li><p>If <var>event</var> equals <var>event id</var>,
+        set <var>found</var> to true.
+
+       <li><p>Prepend <var>event object</var> to <var>filtered events</var>.
+      </ol>
+
+      <li><p>If <var>found</var> is false,
+       return <a>error</a> with <a>error code</a> <a>invalid event id</a>.
+
+      <li>Set property <var>events</var> in <var>body</var> to <var>filtered events</var>.
+     </ol>
+
+ <li><p>Let <var>status</var> be the value of <a>getting a property</a> named <var>status</var> from <var>body</var>.
+
+ <li><p>Let <var>cache timeout</var> be <a><code>null</code></a>.
+
+ <li><p>If <var>status</var> is "<code>queueing</code>" or "<code>creating</code>",
+  then set <var>cache timeout</var> to 1,000
+  or an implementation-defined predicted interval to the next event.
+
+ <li><p><a>Set a property</a> on <var>body</var> with name <var>cacheTimeout</code> and value <var>cache timeout</var>.
+
+  <p class=note>Local ends may use the <var>cacheTimeout</var> property to hint when to next poll for status.
+
+ <li><p>Return success with data <var>body</var>.
+</ol>
+</section> <!-- /Async New Session Status -->
+
+<section>
+<h3><dfn>Cancel New Session</dfn></h3>
+
+<table class="simple jsoncommand">
+ <tr>
+  <th>HTTP Method</th>
+  <th>URI Template</th>
+ </tr>
+ <tr>
+  <td>DELETE</td>
+  <td>/session/async/{<var>session creation job id</var>}</td>
+ </tr>
+</table>
+
+<p>The <a>remote end steps</a> are:
+
+<ol>
+ <li><p>Let <var>session creation job</var> be the job in <a>session creation jobs</a>
+  with a <a>session creation job id</a> matching <var>session creation job id</var>.
+
+ <li><p>If <var>session creation job</var> is <a><code>null</code></a>,
+  return <a>error</a> with <a>error code</a> <a>invalid session creation job id</a>.
+
+ <li><p>If <var>session creation job</var> has a <a data-lt="session creation status">status</a> of
+  "<code>cancelled</code>" or "<code>failed</code>",
+  return <a>error</a> with <a>error code</a> <a>invalid session creation job id</a>.
```

Comment:
Seems like the user-observable response would be the same as having deleted the job, no? If that's the case, should we consider returning "success with data null"?